src/sets.jl

```diff
@@ -73,3 +73,60 @@ function LinearAlgebra.norm(::AbstractVector{<:AbstractJuMPScalar})
           "`@constraint(model, norm(x) <= t)` should now be written as ",
           "`@constraint(model, [t; x] in SecondOrderCone())`")
 end
+
+"""
+    SOS1
+
+SOS1 (Special Ordered Sets type 1) object than can be used to constrain a 
+vector `x` to a set where at most 1 variable can take a non-zero value, all
+others being at 0. 
+The `weights`, when specified, induce an ordering of the variables; as such, 
+they should be unique values. The *k*th element in the set corresponds to the
+*k*th weight in `weights`. See [here](http://lpsolve.sourceforge.net/5.5/SOS.htm)
+for a description of SOS constraints and their potential uses.
+This is a shortcut for the `MathOptInterface.SOS1` set.
+"""
+struct SOS1 <: AbstractVectorSet 
+    weights::Vector{Float64}
+    function SOS1(weights::AbstractVector = Float64[])
```

Comment:
Why providing a default constructor ? It will be weird for the user to get an error "Weight vector in SOS1 is not of length ..." when doing
```
@constraint(model, x in SOS1())
```
since they do not provide any length vector

Reply:
In `moi_set`, if the weight vector is empty, a default one will be created.

So you can go
```Julia
model = Model()
@variable(model, x[1:3])
@constraint(model, x in SOS1())  # Good
@constraint(model, x in SOS1([1, 2, 3]))  # Good
@constraint(model, x in SOS1([1, 2, 3, 4]))  # Errors!
```